tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1781,6 +1840,246 @@ static void test_tx_priv_multi_net_id(void)
+	ASSERT_OK_MSG(bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_ENABLED),
+		      "Failed to Enable gatt proxy");
+	ASSERT_OK_MSG(bt_mesh_subnet_del(TEST_NET_IDX2), "Failed to delete subnet");
+	ASSERT_OK_MSG(bt_mesh_subnet_del(TEST_NET_IDX3), "Failed to delete subnet");
+
+	PASS();
+}
+
+static const struct bt_mesh_test_cfg solicit_trigger_cfg = {
+	.addr = 0x0003,
+	.dev_key = { 0x03 },
+};
+
+static void test_tx_proxy_adv_solicit_trigger(void)
+{
+	tx_proxy_adv_common_init(PROXY_ADV_MULTI_SUBNET_COEX_WAIT_TIME, &solicit_trigger_cfg);
+	ASSERT_OK_MSG(bt_mesh_subnet_add(TEST_NET_IDX2, test_net_key_2),
+		      "Failed to add second subnet");
+
+	k_sleep(UNTIL_UPTIME(PROXY_ADV_MULTI_CHECKPOINT_3));
+
+	/* Solicit first and second network */
+	ASSERT_OK_MSG(bt_mesh_proxy_solicit(TEST_NET_IDX1),
+		      "Failed to start solicitation");
+	ASSERT_OK_MSG(bt_mesh_proxy_solicit(TEST_NET_IDX2),
+		      "Failed to start solicitation");
+
+	PASS();
+}
+
+static void test_rx_proxy_adv_multi_subnet_coex(void)
+{
+	rx_priv_common_init(PROXY_ADV_MULTI_SUBNET_COEX_WAIT_TIME);
+	pp_netkey_ctx_init(&pp_net1);
+	pp_netkey_ctx_init(&pp_net2);
+
+	struct netkey_ctx *nets[] = {&pp_net0, &pp_net1, &pp_net2};
+	struct expected_proxy_adv_evt exp_evt[] = {
+		/** A single subnet is active on the device with GATT Proxy
+		 *  enabled. Verify that the single subnet has exclusive
+		 *  access to the adv medium.
+		 */
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 0, .evt_cnt = 19,
+		 .time = {.after = 0, .before = PROXY_ADV_MULTI_CHECKPOINT_1}},
+
+		/** Two additional subnets are added to the device.
+		 *  Check that the subnets are sharing the adv medium,
+		 *  advertising NET_ID beacons.
+		 */
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 0, .evt_cnt = 8,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_1,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_2}},
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 1, .evt_cnt = 8,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_1,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_2}},
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 2, .evt_cnt = 8,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_1,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_2}},
+
+		/** The second subnet enables Node Identity. Check that NODE_ID
+		 *  is advertised by this subnet, and that the two others
+		 *  continues to advertise NET_ID.
+		 */
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 0, .evt_cnt = 17,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_2,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_3}},
+		{.evt_type = BEACON_TYPE_NODE_ID, .net_idx = 1, .evt_cnt = 17,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_2,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_3}},
+		{.evt_type = BEACON_TYPE_NET_ID, .net_idx = 2, .evt_cnt = 17,
+		 .time = {.after = PROXY_ADV_MULTI_CHECKPOINT_2,
+			  .before = PROXY_ADV_MULTI_CHECKPOINT_3}},
+
+		/** The first and second subnet gets solicited. Check that
+		 *  PRIVATE_NET_ID is advertised by these subnet,
+		 */
```

Comment:
This is again against this paragraph:
> When the server starts advertising with Private Network Identity as a result of the reception of a Solicitation PDU (see Section 6.9.1), the server **shall only** advertise the subnet corresponding to the network key that is used to authenticate the Solicitation PDU; otherwise, the server shall interleave the advertising of each of its subnets. The duration of the advertising is not limited.

The word `only` makes me feel that only one subnet can be advertised at a time and the last solicited subnet "cancels" previous solicitation request. @omkar3141 , can you clarify this?

Reply:
And this phrase I couldn't find, but now I found it (section 7.2.2.2.1):
> A node may advertise with Node Identity or Private Node Identity regardless of the Proxy feature being supported or enabled

Which means either node id or network id, not both

Reply:
About "only":

The phrasing "shall only advertise" here means: Advertise for all of those for which solicitation PDU is received. This phrasing is used at multiple places in the spec to emphasize that, one-to-one relationship between the netID advs for a given subnet and solicited subnet.

You can have following situation that should work properly for network users:

* Assume the following: A node has multiple NetKeys, users of the network are given seperate network keys depending on their access levels. 
* Now, if all users gather at same place and want to connect via same proxy (provided sufficient connections number of connections are supported on that node), they would send solicitation PDU and should be able to connect. Not advertising PrivNetID for each solicited subnet would break the behavior for some users.

Reply:
Ah, sorry, this sentence has different meaning. You can ignore that comment. The question about advertising (private) node id and private network id at the same time is still open...

Reply:
Discussed offline with @omkar3141. Regarding "only" word in that paragraphs: @omkar3141 will file an erratum. For now lets leave this as is and allow to solicit multiple subnets at the same time. Regarding interleaving net id and node id: lets allow it as well. They are 2 different features and the spec doesn't states clearly that this is not allowed.